src/PollingBlockTracker.ts

```diff
@@ -118,37 +117,23 @@ export class PollingBlockTracker
     // return if available
     if (this._currentBlock) {
       return this._currentBlock;
+    } else if (this.#pendingLatestBlock) {
+      return await this.#pendingLatestBlock.promise;
     }
-    // wait for a new latest block
-    const latestBlock: string = await new Promise((resolve, reject) => {
-      // eslint-disable-next-line prefer-const
-      let onLatestBlockUnavailable: InternalListener;
-      const onLatestBlockAvailable = (value: string | PromiseLike<string>) => {
-        this.#removeInternalListener(onLatestBlockAvailable);
-        this.#removeInternalListener(onLatestBlockUnavailable);
-        this.removeListener('error', onLatestBlockUnavailable);
-        resolve(value);
-      };
-      onLatestBlockUnavailable = () => {
-        // if the block tracker is no longer running, reject
-        // and remove the listeners
-        if (!this._isRunning) {
-          this.#removeInternalListener(onLatestBlockAvailable);
-          this.#removeInternalListener(onLatestBlockUnavailable);
-          this.removeListener('latest', onLatestBlockAvailable);
-          this.removeListener('error', onLatestBlockUnavailable);
-          reject(
-            new Error('Block tracker ended before latest block was available'),
-          );
-        }
-      };
-      this.#addInternalListener(onLatestBlockAvailable);
-      this.#addInternalListener(onLatestBlockUnavailable);
-      this.once('latest', onLatestBlockAvailable);
-      this.on('error', onLatestBlockUnavailable);
+
+    const { promise, resolve, reject } = createDeferredPromise<string>({
+      suppressUnhandledRejection: true,
     });
-    // return newly set current block
-    return latestBlock;
+    this.#pendingLatestBlock = { promise, reject };
+
+    // wait for a new latest block
+    const onLatestBlock = (value: string) => {
+      this.#removeInternalListener(onLatestBlock);
+      resolve(value);
+    };
+    this.#addInternalListener(onLatestBlock);
+    this.once('latest', onLatestBlock);
```

Comment:
Do we need to ensure that `#pendingLatestBlock` gets unset at the end of this method?

Reply:
Yes I think so, good catch

Reply:
Done in 83c5de095839056a3669547d8f33f1a23e224d3a